foundations/javascript_basics/clean_code.md

```diff
@@ -42,63 +47,212 @@ const sumArray = function(array) {
+
+#### Use a consistent vocabulary
+
+Variables of the same type should have consistent naming. Consider the following examples from a game:
+
+~~~javascript
+ // Good
+function getPlayerScore();
+function getPlayerName();
+function getPlayerTag();
+
+// Bad
+function getUserScore();
+function fetchPlayerName();
+function retrievePlayer1Tag();
+~~~
+
+In the bad example, three different names are used to refer to the player and the actions taken. Additionally, three different verbs are used to describe these actions. The good example maintains consistency in both variable naming and the verbs used.
+
+Variables should always begin with a noun or an adjective (that is, a noun phrase) and functions with a verb.
+
+Another set of examples can illustrate why this matters:
+
+~~~javascript
+// Good
+const numberOfThings = 10;
+const myName = "Thor";
+const selected = true;
+
+// Bad (these start with verbs, could be confused for functions)
+const getCount = 10;
+const isSelected = true;
+
+// Good
+function getCount() {
+  return numberOfThings;
+}
+
+// Bad (it's a noun)
+function myName() {
+  return "Thor";
+}
+~~~
+
+### Use searchable and immediately understandable names
+
+Sometimes, it can be tempting to use an undeclared variable. Let's take another look at an example:
+
+~~~javascript
+setTimeout(stopTimer, 3600000);
+~~~
+
+The problem is obvious. What does the undeclared variable `3600000` mean and how long is this timeout going to count down before executing `stopTimer`? Even if you know that JavaScript understands time in milliseconds, a calculator is needed.
+
+Now, let's make this code more meaningful by introducing a descriptive variable:
+
+~~~javascript
+const MILLISECONDS_PER_HOUR = 60 * 60 * 1000; // 3,600,000;
+
+setTimeout(stopTimer, MILLISECONDS_PER_HOUR);
+~~~
 
-The second one, however, is much easier to follow.  Even if you don't know exactly what everything in the code is doing, things are named clearly enough that you could guess, and the indentation is consistent enough that it's easy to parse the different parts of the function.
+Much better, isn't it? The variable is declared with a descriptive name, and you don't need to perform any calculations when reading this code. 
 
-There are many different opinions on what constitutes great JavaScript code.  The most important thing is just that you're consistent.  The war between coders that use tabs and coders that use spaces to indent their code is so engrained that [it's essentially a joke by now](https://www.youtube.com/watch?v=SsoOG6ZeyUI), but it doesn't _really_ matter as long as you're consistent.
+You might wonder why this variable is declared with all caps. This is a convention to be used when the programmer is absolutely sure that the variable is _truly_ a constant. We know that the milliseconds in an hour will never change, so it's appropriate here.
 
-### Rules of Thumb
+#### About camelCase
```

Comment:
Not a big deal but it feels like cameCase can probably be mentioned earlier since it is used throughout this reading.